CIPs/cip-0025.md

```diff
@@ -42,22 +42,28 @@ If block.number >= FORK_HEIGHT (TBC), add a precompiled contract for Ed25519 sig
 
 The proposal adds a new precompiled function `ED25519VFY` with the following input and output.
 
-`ED25519VFY` takes as input 128 octets:
+`ED25519VFY` takes as input:
 
-- message: the 32-octet message that was signed
 - public key: the 32-octet Ed25519 public key of the signer
 - signature: the 64-octet Ed25519 signature
+- message: the arbitrary length message that was signed
 
-`ED25519VFY` returns as output 4 octets:
+`ED25519VFY` returns as output 32 octets:
 
-- `0x00000000` if signature is valid
+- `0` if signature is valid
 - any non-zero value indicates a signature verification failure
 
 ### Address
 The address of `ED25519VFY` is `TBC`.
 
 ### Gas costs
-Gas cost for `ED25519VFY` is 2000.
+Gas cost for `ED25519VFY` is calculated with the formula:
+```
+words = len(INPUT) / SHA2_512_WORD_LENGTH
+gas = ED25519VFY_BASE_GAS + SHA2_512_BASE_GAS + (words * SHA2_512_PER_WORD_GAS)
```

Comment:
I made an error here, and this logic I gave you rounds words down when we ought to round up. Here's the new logic I'm using in CIP20
```
func wordMeteredGasPrice(base, perWord, wordSize, inputLength uint64) uint64 {
        // round up to next whole word
        lengthCeil := inputLength + wordSize - 1
        words := lengthCeil / wordSize
        return base + perWord*words
}
```

Reply:
Good catch :+1:

Reply:
(y) @prestwich I've updated the formula, thanks!